version: '3.3'
services:
  configuration_server:
    image: training/configuration
    restart: on-failure
    environment:
      - ENCRYPT_KEY=${ENCRYPT_KEY}
    networks:
      cloud:
        ipv4_address: ${CONFIGURATION_SERVER}

  discovery_server:
    image: training/discovery
    restart: on-failure
    ports:
      - 8001:8080
    environment:
      - ENCRYPT_KEY=${ENCRYPT_KEY}
      - PING_TIMEOUT=${PING_TIMEOUT}
      - CONFIGURATION_SERVER_URI=${CONFIGURATION_SERVER_URI}
      - DEFAULT_PORT=${DEFAULT_PORT}
    networks:
      cloud:
        ipv4_address: ${DISCOVERY_SERVER}

  gateway_server:
    image: training/gateway
    restart: on-failure
    environment:
      - ENCRYPT_KEY=${ENCRYPT_KEY}
      - PING_TIMEOUT=${PING_TIMEOUT}
      - CONFIGURATION_SERVER_URI=${CONFIGURATION_SERVER_URI}
      - DEFAULT_PORT=${DEFAULT_PORT}
    networks:
      cloud:
        ipv4_address: ${GATEWAY_SERVER}

  users_database:
    image: postgres:9.6.5-alpine
    environment:
      - POSTGRES_DB=users
      - POSTGRES_USER=admin
      - POSTGRES_PASSWORD=admin
      - PGDATA=/var/lib/postgresql/data/application
    volumes:
      - ./data/users:/var/lib/postgresql/data/application
    networks:
      cloud:
        ipv4_address: ${USERS_DATABASE_SERVER}

networks:
  cloud:
    driver: bridge
    ipam:
      config:
        - subnet: ${SUBNET}